frontend/lint/eslint-rules/no-unconditional-metabase-links-render.js

```diff
@@ -0,0 +1,223 @@
+}
+
+const ADD_COMMENT_MESSAGE =
+  'add comment to indicate the reason why this rule needs to be disabled.\nExample: "// eslint-disable-next-line no-unconditional-metabase-links-render -- This links only shows for admins."';
+const ERROR_MESSAGE =
+  "Metabase links must be rendered conditionally.\n\nPlease import `getShowMetabaseLinks` selector from `metabase/selectors/whitelabel` and use it to conditionally render Metabase links.\n\nOr " +
+  ADD_COMMENT_MESSAGE;
+const LITERAL_METABASE_URL_REGEX =
+  /(metabase\.com\/docs|metabase\.com\/learn)($|\/)/;
+
+// eslint-disable-next-line import/no-commonjs
+module.exports = {
+  meta: {
+    type: "problem",
+    docs: {
+      description:
+        "Ensure that Metabase links are rendered conditionally using a `show-metabase-links` setting",
+    },
+    schema: [], // no options
+  },
+
+  create(context) {
+    let isMetabaseSettingsImported = false;
+    let isGetDocsUrlSelectorImported = false;
+    let isGetLearnUrlSelectorImported = false;
+    let isGetShowMetabaseLinksSelectorImported = false;
+
+    /**
+     * @typedef DefaultImport
+     * @property {true} isDefault
+     * @property {string} source
+     *
+     * @typedef NamedImport
+     * @property {string} named
+     * @property {string} source
+     *
+     * @param {object} node
+     * @param {DefaultImport|NamedImport} parameter
+     */
+    function checkIfModuleIsImported(node, { isDefault, named, source }) {
+      if (getImportNodeLocation(node) === source) {
+        const variables = context.getDeclaredVariables(node);
+        if (isDefault) {
+          return variables.some(
+            variable => variable.defs[0].node.type === "ImportDefaultSpecifier",
+          );
+        }
+
+        // Named import
+        return variables.some(
+          variable =>
+            variable.defs[0].node.type === "ImportSpecifier" &&
+            variable.name === named,
+        );
+      }
+
+      return false;
+    }
+
+    return {
+      ImportDeclaration(node) {
+        if (
+          checkIfModuleIsImported(node, {
+            isDefault: true,
+            source: "metabase/lib/settings",
+          })
+        ) {
+          isMetabaseSettingsImported = true;
+        }
+        if (
+          checkIfModuleIsImported(node, {
+            named: "getDocsUrl",
+            source: "metabase/selectors/settings",
+          })
+        ) {
+          isGetDocsUrlSelectorImported = true;
+        }
+        if (
+          checkIfModuleIsImported(node, {
+            named: "getLearnUrl",
+            source: "metabase/selectors/settings",
+          })
+        ) {
+          isGetLearnUrlSelectorImported = true;
+        }
+        if (
+          checkIfModuleIsImported(node, {
+            named: "getShowMetabaseLinks",
+            source: "metabase/selectors/whitelabel",
+          })
+        ) {
+          isGetShowMetabaseLinksSelectorImported = true;
+        }
+      },
+      CallExpression(node) {
```

Comment:
Traverse any call expression e.g. `myFunction()`, the good part is that our selectors are expected to always be called. `useSelector(state => getDocsUrl(state, { page: xxx })`. If they were to be passed directly e.g. `useSelector(getDocsUrl)`, it likely won't be traversed (I didn't test).